common/beerocks/bwl/dwpal/ap_wlan_hal_dwpal.cpp

```diff
@@ -416,6 +417,303 @@ static std::shared_ptr<char> generate_client_assoc_event(const std::string &even
     return msg_buff;
 }
 
+// Checks if file exists
+static bool file_exists(const std::string &fname)
+{
+    struct stat st;
+    return (stat(fname.c_str(), &st) == 0);
+}
+
+// Finds value by key in a string vector representing hostapd config (head or VAP)
+// and returns value found in the supplied string object.
+static bool hostapd_config_get_value(const std::vector<std::string> &hostapd_config,
+                                     const std::string &key, std::string &value)
+{
+    std::string key_eq(key + "=");
+    auto it_str =
+        std::find_if(hostapd_config.begin(), hostapd_config.end(), [&](std::string str) -> bool {
+            return (str.compare(0, key_eq.length(), key_eq) == 0);
+        });
+    if (it_str >= hostapd_config.end()) {
+        return false;
+    }
+    value.assign(*it_str, key_eq.length(), std::string::npos);
+    return true;
+}
+
+// Finds a value by key in a string vector representing hostapd config (head or VAP)
+// and replaces it with the supplied one. If the value does not exist, it's added.
+// If the value is empty string it's deleted.
+static bool hostapd_config_set_value(std::vector<std::string> &hostapd_config,
+                                     const std::string &key, const std::string &value)
+{
+    std::string key_eq(key + "=");
+    auto it_str =
+        std::find_if(hostapd_config.begin(), hostapd_config.end(), [&](std::string str) -> bool {
+            return (str.compare(0, key_eq.length(), key_eq) == 0);
+        });
+    // Delete the key if value is empty
+    if (value.length() == 0) {
+        if (it_str >= hostapd_config.end()) {
+            hostapd_config.erase(it_str);
+        }
+        // We are good if it is already not present
+        return true;
+    }
+    // If the key is not found add it
+    if (it_str >= hostapd_config.end() && value.length() > 0) {
+        hostapd_config.push_back(key_eq + value);
+        return true;
+    }
+    // Replace the value
+    it_str->resize(key_eq.length() + value.length());
+    it_str->replace(key_eq.length(), std::string::npos, value);
```

Comment:
This (= the whole function after the it_str declaration) can be simplified to:

```suggestion
if (it_str != hostapd_config.end()) {
    hostapd_config.erase(it_str);
}
if (value) {
    hostapd_config.push_back(key_eq + value);
}
```

Reply:
I'm trying to preserve the order of parameters whenever possible. Are you certain that there are no ordering requirements in hostapd config besides the bss key-value pairs being on top of the VAP sections? I can just remove/add valued if that's the case.

Reply:
There are no ordering requirements in the hostapd.conf file.